src/python/bot/fuzzers/mutator_plugin.py

```diff
@@ -0,0 +1,155 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Getting and using custom mutator plugins for libFuzzer."""
+
+import os
+
+from base import utils
+from google_cloud_utils import storage
+from system import archive
+from system import environment
+from system import shell
+
+MUTATOR_SHARED_OBJECT_FILENAME = 'mutator-plugin.so'
+ARCHIVES_SUBDIR_NAME = 'archives'
+PLUGINS_SUBDIR_NAME = 'plugins'
+
+
+def clear_plugins_directory():
+  shell.remove(environment.get_value('MUTATOR_PLUGINS_DIR'))
+
+
+def _get_mutator_plugins_bucket_url():
+  """Returns the url of the mutator plugin's cloud storage bucket."""
+  return 'gs://%s' % environment.get_value('MUTATOR_PLUGINS_BUCKET')
+
+
+def _get_mutator_plugins_subdir(subdir):
+  """Returns the path of the subdirectory named |subdir| in
+  MUTATOR_PLUGINS_DIR."""
+  return os.path.join(environment.get_value('MUTATOR_PLUGINS_DIR'), subdir)
+
+
+def _get_mutator_plugins_archives_dir():
+  """Returns path of the archives subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(ARCHIVES_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_unpacked_dir():
+  """Returns path of the unpacked plugins subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(PLUGINS_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_from_bucket():
+  """Returns list of the mutator plugin archives in the mutator plugin storage
+  bucket."""
+  return storage.list_blobs(_get_mutator_plugins_bucket_url())
+
+
+def _download_mutator_plugin_archive(mutator_plugin_archive):
+  """Downloads the |mutator_plugin_archive| from the mutator plugin storage
+  bucket to the plugin archives directory. Returns the path archive was
+  downloaded to."""
+  file_path = os.path.join(_get_mutator_plugins_archives_dir(),
+                           mutator_plugin_archive)
+  url = '%s/%s' % (_get_mutator_plugins_bucket_url(), mutator_plugin_archive)
+  assert storage.copy_file_from(url, file_path)
+  return file_path
+
+
+def _unpack_mutator_plugin(mutator_plugin_archive_path):
+  """Unpacks |mutator_plugin_archive_path| in the unpacked plugins directory and
+  returns the path it was unpacked into."""
+  mutator_plugin_name = os.path.basename(
+      os.path.splitext(mutator_plugin_archive_path)[0])
+  unpacked_plugin_dir = os.path.join(_get_mutator_plugins_unpacked_dir(),
+                                     mutator_plugin_name)
+  archive.unpack(mutator_plugin_archive_path, unpacked_plugin_dir)
+  return unpacked_plugin_dir
+
+
+def _extract_name_from_archive(plugin_archive_filename):
+  """Parses |plugin_archive_filename| which should be named using the schema:
+  '$NAME-$JOB-$FUZZ_TARGET.zip'. Returns tuple containing
```

Comment:
Did you see my comment in design doc. i think we need an accompanying yaml or .options file that can have this info. name feels too hacky.